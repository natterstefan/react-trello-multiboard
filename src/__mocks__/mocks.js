import { pick } from 'lodash'

// This file contains mocked responses from the data/trello API.
export const mockExampleBoardConfig = {
  board: 'hello-world',
  estimates_with_round_brackets: true,
  estimates_with_square_brackets: true,
  lists: /#upcoming/,
  shortcut: 'hw',
}

export const mockEampleUser1 = {
  avatarHash: '123456789',
  fullName: 'Example Username',
  id: 'member-1',
  initials: 'EU',
  username: 'exampleusername',
}

export const mockEampleUser2 = {
  avatarHash: '987654321',
  fullName: 'Another User',
  id: 'member-2',
  initials: 'AU',
  username: 'anotheruser',
}

export const mockEampleUser3 = {
  avatarHash: null,
  fullName: 'Third User',
  id: 'member-3',
  initials: 'TU',
  username: 'thirduser',
}

export const mockExampleMemberResponse = username => {
  switch (username) {
    case 'anotheruser':
      return mockEampleUser2
    case 'thirduser':
      return mockEampleUser3
    default:
      return mockEampleUser1
  }
}

export const mockExampleBoardResponse = { name: 'hello-world', id: 'board-1' }

// most likely it should be an array with two lists (one matching and one not)
export const mockExampleListResponse = {
  id: 'list-123',
  idBoard: 'board-1',
  name: 'Test List Name #upcoming',
}

export const mockExampleCardsResponse = {
  id: 'card-1',
  name: 'Test Card Name',
  idBoard: 'board-1',
  idMembers: [mockEampleUser1.id, mockEampleUser2.id],
  idList: 'list-1',
  shortUrl: 'https://trello.com/c/abcdefgh',
  members: [
    pick(mockEampleUser1, ['id', 'username', 'fullName']),
    pick(mockEampleUser2, ['id', 'username', 'fullName']),
  ],
}
