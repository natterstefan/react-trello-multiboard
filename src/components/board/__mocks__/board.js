import {
  mockExampleUser1,
  mockExampleUser2,
  mockExampleBoardResponse,
  mockExampleListResponse,
} from '../../../__mocks__/mocks'

export const props = {
  board: {
    id: 'board-1',
  },
}

export const storeStateMock = {
  boards: {
    data: [
      {
        board: mockExampleBoardResponse,
      },
      {
        board: {
          id: 'board-2',
          name: 'another-example-board',
        },
      },
    ],
  },
  lists: {
    'board-1': {
      isLoading: false,
      data: [mockExampleListResponse],
      error: null,
    },
  },
  info: {
    version: 1,
  },
  app: {
    memberToggle: {
      togglePreferred: false,
      togglePreferredMember: null,
    },
  },
  members: {
    members: [mockExampleUser1, mockExampleUser2],
  },
}
