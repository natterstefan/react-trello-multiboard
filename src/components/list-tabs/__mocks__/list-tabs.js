export const storeStateMock = {
  app: {
    config: {
      lists: ['list-1', 'list-2'],
    },
    listToggle: {
      toggleList: 'list-1',
    },
  },
}
