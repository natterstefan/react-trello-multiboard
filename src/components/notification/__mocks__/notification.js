export const props = { message: 'some notification' }

export const storeStateMock = {
  boards: {
    isLoading: false,
    error: null,
  },
  lists: {
    'board-1': {
      isLoading: false,
      error: null,
    },
  },
  members: {
    isLoading: false,
    error: null,
  },
}

export const storeWithBoardError = {
  boards: {
    isLoading: false,
    error: {
      status: 401,
      responseText: 'example response',
    },
  },
}

export const storeWithAnyError = {
  user: {
    error: { message: 'some error user message ' },
  },
}
