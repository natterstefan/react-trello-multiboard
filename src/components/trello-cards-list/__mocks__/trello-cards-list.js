import {
  mockExampleCardsResponse,
  mockExampleBoardConfig,
  mockExampleListResponse,
} from '../../../__mocks__/mocks'

export const props = {
  list: mockExampleListResponse,
  config: mockExampleBoardConfig,
}

export const storeStateMock = {
  cards: {
    'list-1': {
      data: [
        {
          card: mockExampleCardsResponse,
          config: mockExampleBoardConfig,
        },
      ],
      isLoading: true,
      error: null,
    },
  },
}
