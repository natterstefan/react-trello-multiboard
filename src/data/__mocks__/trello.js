// JEST - How to mock modules
//
// either we mock it in each test seperately
// - >>> https://github.com/facebook/jest/issues/2582#issuecomment-346886018
// - https://stackoverflow.com/a/40465435/1238150
//
// or provide a solution for every test in the __mocks__
// - https://facebook.github.io/jest/docs/en/manual-mocks.html
import {
  mockExampleMemberResponse,
  mockExampleBoardResponse,
  mockExampleListResponse,
  mockExampleCardsResponse,
} from '../../__mocks__/mocks'

// NOTE: use mockResolvedValue instead
export const authenticateUser = jest.fn().mockResolvedValue({})
export const getBoard = jest.fn().mockResolvedValue(mockExampleBoardResponse)
export const getMeBoards = jest.fn().mockResolvedValue([mockExampleBoardResponse])
export const getLists = jest.fn().mockResolvedValue([mockExampleListResponse])
export const getCards = jest.fn().mockResolvedValue([mockExampleCardsResponse])
export const getMember = jest.fn(username => Promise.resolve(mockExampleMemberResponse(username)))
export const isTrelloAvailable = jest.fn().mockReturnValue(true)

export default {
  authenticateUser,
  getBoard,
  getMeBoards,
  getLists,
  getCards,
  getMember,
  isTrelloAvailable,
}
