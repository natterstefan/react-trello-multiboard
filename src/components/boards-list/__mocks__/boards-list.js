import { mockExampleUser1, mockExampleBoardResponse } from '../../../__mocks__/mocks'

export const storeStateMock = {
  boards: {
    data: [{ board: mockExampleBoardResponse }],
    error: null,
    isLoading: true,
  },
  app: {
    memberToggle: {
      togglePreferred: false,
      togglePreferredMember: null,
    },
  },
  members: { members: [mockExampleUser1] },
}
