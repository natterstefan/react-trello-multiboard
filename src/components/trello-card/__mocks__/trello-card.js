import { mockEampleUser1, mockEampleUser2 } from '../../../__mocks__/mocks'

export const props = {
  addEstimations: jest.fn(),
  boardName: 'example-board',
  card: {
    badges: {
      description: false,
      comments: 1,
      attachments: 1,
      checkItems: 2,
      checkItemsChecked: 1,
    },
    id: 'id-1',
    idBoard: 'board-1',
    idMembers: ['member-1', 'member-2'],
    members: [mockEampleUser1, mockEampleUser2],
    name: 'example-name',
    shortUrl: 'https://trello.com/id-1',
  },
  config: {
    estimates_with_round_brackets: false,
    estimates_with_square_brackets: false,
  },
  isHidden: false,
  listName: 'example-list',
}

export const storeStateMock = {
  boards: {
    data: [
      {
        board: {
          // will be added to the TrelloCard (because it matches the idBoards above)
          id: 'board-1',
          name: 'example-board',
        },
      },
      {
        board: {
          id: 'board-2',
          name: 'another-example-board',
        },
      },
    ],
  },
  app: {
    memberToggle: {
      togglePreferred: false,
      togglePreferredMember: null,
    },
  },
  members: {
    members: [],
  },
}

export const storeStateMockPreferred = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: null,
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}

export const storeStateMockPreferredMember = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: 'member-1',
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}

export const storeStateMockPreferredMemberInvalid = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: 'member-3',
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}
