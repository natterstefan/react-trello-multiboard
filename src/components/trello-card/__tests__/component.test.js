import React from 'react'
import { mount, shallow } from 'enzyme'
import { merge } from 'lodash'

import TrelloCard from '../component'
import TrelloCardUi from '../card'

import { props } from '../__mocks__/props'

describe('Component/TrelloCard', () => {
  test('should render without throwing an error', () => {
    expect(mount(<TrelloCard {...props} />)).toMatchSnapshot()
  })

  test('should render with display:none when isHidden', () => {
    const wrapper = shallow(<TrelloCard {...props} isHidden />)
    const div = wrapper.find('div.card')
    expect(div.length).toBe(1)
    expect(div.first().prop('style')).toMatchObject({ display: 'none' })
  })

  test('should calculate the member classnames properly', () => {
    const wrapper = shallow(<TrelloCard {...props} />)
    const instance = wrapper.instance()
    expect(instance.getMemberClass()).toBe('member_member-1 member_member-2')
  })

  test('should be selectable by class "member_member-1"', () => {
    expect(shallow(<TrelloCard {...props} />).is('.member_member-1')).toBe(true)
  })

  test('should contain a div container, where a TrelloCardUi is rendered into', () => {
    const wrapper = mount(<TrelloCard {...props} />)
    expect(wrapper.find('div#card_id-1')).toHaveLength(1)
    expect(wrapper.find(TrelloCardUi)).toHaveLength(1)
  })

  test('should add default estimations to redux by invoking addEstimations prop function', () => {
    const addEstimations = jest.fn()
    shallow(<TrelloCard {...props} addEstimations={addEstimations} />)
    expect(addEstimations).toHaveBeenCalledTimes(1)
    expect(addEstimations).toBeCalledWith({ consumed: 0, estimated: 0 })
  })

  test('should add estimations to it addEstimations prop function', () => {
    const newProps = merge({}, props, {
      addEstimations: jest.fn(),
      card: {
        name: 'example-name (5)[1]',
      },
      config: {
        estimates_with_round_brackets: true,
        estimates_with_square_brackets: true,
      },
    })
    shallow(<TrelloCard {...newProps} />)
    expect(newProps.addEstimations).toHaveBeenCalledTimes(1)
    expect(newProps.addEstimations).toBeCalledWith({ consumed: 1, estimated: 5 })
  })

  test('should add only valid estimations to it addEstimations prop function', () => {
    const newProps = merge({}, props, {
      addEstimations: jest.fn(),
      card: {
        name: 'example-name (5)[xx]',
      },
      config: {
        estimates_with_round_brackets: true,
        estimates_with_square_brackets: true,
      },
    })
    shallow(<TrelloCard {...newProps} />)
    expect(newProps.addEstimations).toHaveBeenCalledTimes(1)
    expect(newProps.addEstimations).toBeCalledWith({ consumed: 0, estimated: 5 })
  })

  test('should add only valid estimations to it addEstimations prop function', () => {
    console.error = jest.fn() // disable propTypes warning
    const newProps = merge({}, props, {
      addEstimations: jest.fn(),
      card: {
        name: null,
      },
      config: {
        estimates_with_round_brackets: true,
        estimates_with_square_brackets: true,
      },
    })
    shallow(<TrelloCard {...newProps} />)
    expect(newProps.addEstimations).toHaveBeenCalledTimes(1)
    expect(newProps.addEstimations).toBeCalledWith({ consumed: 0, estimated: 0 })
  })
})
