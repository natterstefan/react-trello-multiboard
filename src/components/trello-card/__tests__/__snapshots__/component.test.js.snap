// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/TrelloCard should render without throwing an error 1`] = `
<TrelloCard
  addEstimations={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "consumed": 0,
            "estimated": 0,
          },
        ],
      ],
    }
  }
  boardName="example-board"
  card={
    Object {
      "badges": Object {
        "attachments": 1,
        "checkItems": 2,
        "checkItemsChecked": 1,
        "comments": 1,
        "description": false,
      },
      "id": "id-1",
      "idBoard": "board-1",
      "idMembers": Array [
        "member-1",
        "member-2",
      ],
      "members": Array [
        Object {
          "avatarHash": "123456789",
          "fullName": "Example Username",
          "id": "member-1",
          "initials": "EU",
          "username": "exampleusername",
        },
        Object {
          "avatarHash": "987654321",
          "fullName": "Another User",
          "id": "member-2",
          "initials": "AU",
          "username": "anotheruser",
        },
      ],
      "name": "example-name",
      "shortUrl": "https://trello.com/id-1",
    }
  }
  config={
    Object {
      "estimates_with_round_brackets": false,
      "estimates_with_square_brackets": false,
    }
  }
  isHidden={false}
  listName="example-list"
>
  <div
    className="card member_member-1 member_member-2"
    id="card_id-1"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <TrelloCardUI
      badges={
        Object {
          "attachments": 1,
          "checkItems": 2,
          "checkItemsChecked": 1,
          "comments": 1,
          "description": false,
        }
      }
      boardName="example-board"
      id="id-1"
      idBoard="board-1"
      idMembers={
        Array [
          "member-1",
          "member-2",
        ]
      }
      listName="example-list"
      members={
        Array [
          Object {
            "avatarHash": "123456789",
            "fullName": "Example Username",
            "id": "member-1",
            "initials": "EU",
            "username": "exampleusername",
          },
          Object {
            "avatarHash": "987654321",
            "fullName": "Another User",
            "id": "member-2",
            "initials": "AU",
            "username": "anotheruser",
          },
        ]
      }
      name="example-name"
      shortUrl="https://trello.com/id-1"
    >
      <a
        className="trello-card-wrapper"
        href="https://trello.com/id-1"
      >
        <div
          className="trello-card"
        >
          <div
            className="trello-card__content"
          >
            <div
              className="trello-card__title"
            >
              example-name
            </div>
            <div
              className="trello-card__badges"
            >
              <div>
                <FontAwesomeIcon$1
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Array [
                      "fa",
                      "comment-dots",
                    ]
                  }
                  listItem={false}
                  mask={null}
                  name=""
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="sm"
                  spin={false}
                  symbol={false}
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-comment-dots fa-w-16 fa-sm "
                    data-icon="comment-dots"
                    data-prefix="fa"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32zM128 272c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon$1>
                 
                1
              </div>
              <div>
                <FontAwesomeIcon$1
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Array [
                      "fa",
                      "paperclip",
                    ]
                  }
                  listItem={false}
                  mask={null}
                  name=""
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="sm"
                  spin={false}
                  symbol={false}
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-paperclip fa-w-14 fa-sm "
                    data-icon="paperclip"
                    data-prefix="fa"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M43.246 466.142c-58.43-60.289-57.341-157.511 1.386-217.581L254.392 34c44.316-45.332 116.351-45.336 160.671 0 43.89 44.894 43.943 117.329 0 162.276L232.214 383.128c-29.855 30.537-78.633 30.111-107.982-.998-28.275-29.97-27.368-77.473 1.452-106.953l143.743-146.835c6.182-6.314 16.312-6.422 22.626-.241l22.861 22.379c6.315 6.182 6.422 16.312.241 22.626L171.427 319.927c-4.932 5.045-5.236 13.428-.648 18.292 4.372 4.634 11.245 4.711 15.688.165l182.849-186.851c19.613-20.062 19.613-52.725-.011-72.798-19.189-19.627-49.957-19.637-69.154 0L90.39 293.295c-34.763 35.56-35.299 93.12-1.191 128.313 34.01 35.093 88.985 35.137 123.058.286l172.06-175.999c6.177-6.319 16.307-6.433 22.626-.256l22.877 22.364c6.319 6.177 6.434 16.307.256 22.626l-172.06 175.998c-59.576 60.938-155.943 60.216-214.77-.485z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon$1>
                 
                1
              </div>
              <div>
                <FontAwesomeIcon$1
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Array [
                      "fa",
                      "check-square",
                    ]
                  }
                  listItem={false}
                  mask={null}
                  name=""
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="sm"
                  spin={false}
                  symbol={false}
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-check-square fa-w-14 fa-sm "
                    data-icon="check-square"
                    data-prefix="fa"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon$1>
                 
                1
                /
                2
              </div>
            </div>
            <div
              className="trello-card__members"
            >
              <ul>
                <li
                  key="https://trello.com/id-1-123456789"
                >
                  <div
                    className="trello-card__member"
                  >
                    <img
                      alt="Example Username"
                      className="trello-card__member--image"
                      src="https://trello-avatars.s3.amazonaws.com/123456789/50.png"
                    />
                  </div>
                </li>
                <li
                  key="https://trello.com/id-1-987654321"
                >
                  <div
                    className="trello-card__member"
                  >
                    <img
                      alt="Another User"
                      className="trello-card__member--image"
                      src="https://trello-avatars.s3.amazonaws.com/987654321/50.png"
                    />
                  </div>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="trello-card__board-details"
          >
            <div
              className="trello-card__board-name"
            >
              example-board
              :
            </div>
            <div
              className="trello-card__list-name"
            >
              example-list
            </div>
          </div>
        </div>
      </a>
    </TrelloCardUI>
  </div>
</TrelloCard>
`;
